Run test suite via tox + Travis CI

sevein opened the pull request:
- Added tox envs: `py27`, `py36`, `py27-flake8`, `py36-flake8`.
- Execute all envs in Travis CI.

This closes #612.

sevein committed:
Run test suite via tox + Travis CI

Run AM test suite via tox and wire it in Travis CI.

Both py27 and py36 (allowed to fail).

This closes #612.

sevein committed:
Fix test in dashboardsetting_manager

helenst commented:
Looks good in terms of actually running the tests. Do we want to fix them all before merging this?

sevein commented:
> Do we want to fix them all before merging this?

Yes, I'll give it a try today. It shouldn't be too hard but I'm surprised they broke. Thanks!

sevein commented on tox.ini:
```diff
@@ -1,14 +1,24 @@
+    pip install -r src/archivematicaCommon/requirements/test.txt
+    pip install -r src/MCPClient/requirements/test.txt
+    pip install -r src/dashboard/src/requirements/test.txt
+    py.test {posargs}
```
`{posargs}` is used here so we can pass arguments to pytest, e.g.:

- `tox -e archivematica -- -x`
- `tox -e archivematica -- src/dashboard`
- [etc...](https://docs.pytest.org/en/latest/usage.html)

sevein committed:
Update requests, fixes tests

vcrpy 1.11.1 made some changes only compatible with recent versions of
requests. I've checked the changelog of requests, it doesn't seem that
backward-incompatible changes were introduced. We should be safe!

sevein committed:
MCPClient: fix test_json_conversion

sevein commented:
@jrwdunham @helenst, tests fixed! 😅

helenst reviewed (approved):
Nice work @sevein !

jrwdunham reviewed (approved):
Yay! Happy day.
